### PR TITLE
Exception formatting

### PR DESCRIPTION
**Pull request by Austin-Tan**
Partially addresses #1043.

## Changes

Thought I would parcel out this issue across a few PRs since it requires touching so many files, and I want to make sure I'm heading in the right direction.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Jaeger/ApacheThrift/TException.cs [hunk 1]
```diff
@@ -32,5 +33,10 @@ public TException(string message, Exception inner)
             : base(message, inner)
         {
         }
+
+        // A constructor is needed for serialization when an
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Exporter.Jaeger/ApacheThrift/TException.cs:37]: Agree with the comment, but these Thrift Exceptions aren't being propagated are they?

Comment (Austin-Tan) [reply src/OpenTelemetry.Exporter.Jaeger/ApacheThrift/TException.cs:37]: I don't think they are. Added the seriazlization following [guide @reyang linked](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/exceptions/creating-and-throwing-exceptions#defining-exception-classes) in #1043 but I see now that we were mostly referring to that for exception message consistency. I'll remove this unless there are other thoughts

File: src/OpenTelemetry.Shims.OpenTracing/SpanShim.cs [hunk 2]
```diff
@@ -45,11 +45,11 @@ internal sealed class SpanShim : global::OpenTracing.ISpan
 
         public SpanShim(TelemetrySpan span)
         {
-            this.Span = span ?? throw new ArgumentNullException(nameof(span));
+            this.Span = span ?? throw new ArgumentNullException(nameof(span), "Parameter cannot be null");
```
Comment (eddynaka) [inline src/OpenTelemetry.Shims.OpenTracing/SpanShim.cs:48]: should we end the text with a .?

File: src/OpenTelemetry.Exporter.Jaeger/ApacheThrift/TException.cs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿// <auto-generated/> (Turns off StyleCop analysis in this file.)
+// <auto-generated/> (Turns off StyleCop analysis in this file.)
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.Jaeger/ApacheThrift/TException.cs:1]: we dont need to touch these files as they are auto generated.

File: src/OpenTelemetry.Exporter.Jaeger/ApacheThrift/TException.cs (hunk 1 above)
Comment (cijothomas) [reply src/OpenTelemetry.Exporter.Jaeger/ApacheThrift/TException.cs:37]: lets leave these files untouched

File: src/OpenTelemetry.Shims.OpenTracing/SpanShim.cs (hunk 2 above)
Comment (cijothomas) [reply src/OpenTelemetry.Shims.OpenTracing/SpanShim.cs:48]: **https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/exceptions/creating-and-throwing-exceptions**
From this guidelines, there is no "." ending.

**Comment by Austin-Tan** [comment]
Will downsize scope of PR so we can pass it easily on a couple of exception changes by EOD

**Pull request closed without merging**